src/WebJobs.Extensions.DurableTask/LinuxAppServiceLogger.cs

```diff
@@ -129,9 +227,54 @@ private string GenerateJsonStr(EventWrittenEventArgs eventData)
-            jsonString = jsonString.Replace("\n", "\\n").Replace("\r", "\\r");
+            string jsonString = "";
+            if (!this.writeToConsole)
+            {
+                // This path supports the legacy regex-based parser in Linux Dedicated.
+                // In the future, we'll be able to remove this and use JSON logging only
+                IList<string> jsonKeys = json.Properties().Select(p => p.Name).ToList();
+                List<string> regexValues = new List<string>();
+
+                foreach (string column in this.orderedRegexCol)
+                {
+                    string val = "";
+                    if (json.TryGetValue(column, out JToken valJToken))
+                    {
+
+                        // We escape a few special characters to avoid parsing problems:
+                        // (1) Escaping newline-like characters such as \n and \r to keep logs being 1 line
+                        // (2) Escaping double-quotes (") to be single-quotes (') because some fields in our regex string are
+                        //     deliniated by double-quotes. Note, this was a convention copied from the Functions Host regex
+                        //     which also uses double-quotes to capture columns that may contain commas inside them.
+                        // (3) Escaping commas (",") for ";;" because commas separate our columns and so they have the potential to
+                        //     disrupt parsing.
+                        // Note: In retrospective, perhaps the regex-string could have been designed to avoid these awkward
+                        // parsing problems, but it wasn't because (1) it followed conventions from other regex-strings in our system
+                        // and because we asssumed we'd have a JSON-based logger that would have avoided these problems.
+                        val = (string)valJToken;
+                        val = val.Replace("\n", "\\n").Replace("\r", "\\r").Replace("\"", "'").Replace(",", ";;");
+                    }
+
+                    if (column == "Details" || column == "ExceptionMessage")
+                    {
+                        // Since Details and Exceptions may include commas, our regex string
+                        // expects this field to be wrapped in double-quotes to avoid capturing an "inner comma",
+                        // a convention adopted by the Functions Host regex string.
+                        val = '"' + val + '"';
+                    }
+
+                    regexValues.Add(val);
+                }
+
+                jsonString = string.Join(",", regexValues);
+            }
+            else
+            {
+                // Generate string-representation of JSON. Also remove newlines.
+                jsonString = json.ToString(Newtonsoft.Json.Formatting.None);
+                jsonString = jsonString.Replace("\n", "\\n").Replace("\r", "\\r");
```

Comment:
Is this last step necessary? I did a quick test an it seems like `JObject.ToString()` already replaces newline characters with `\\n`, etc. If we don't already, it would be great to have a test that confirms the correct behavior.

Reply:
I tested that too and, indeed, we can remove it